src/go/k8s/controllers/redpanda/cluster_controller_configuration.go

```diff
@@ -342,6 +343,9 @@ func (r *ClusterReconciler) synchronizeStatusWithCluster(
 		if err := r.Status().Update(ctx, redpandaCluster); err != nil {
 			return nil, errorWithContext(err, "could not update condition on cluster")
 		}
+		if err := statefulset.MarkPodsForUpdate(ctx); err != nil {
+			return nil, errorWithContext(err, "could not mark pods for update")
+		}
```

Comment:
Not all configuration changes requires Pod restart. There is specific logic that triggers pod restarts. Should we always set that condition?

Reply:
Are you asking if we should always set the condition but set the status of the condition to false if the pod doesn't need a restart?

Perhaps, but that's out of scope for this issue.

Reply:
With this change every configuration change will cause Pod restart, right? If yes, then it's a regression in the reconciliation logic.

Reply:
no. it will not. We have e2e tests for that.

Reply:
Well.. it does look like it should, but we do have tests and it passed. Will take another look.

Reply:
Ok, does this look right to you now?

Reply:
Yes, thank you